app/models/channel.rb

```diff
@@ -48,6 +50,7 @@ class Channel < ApplicationRecord
   validate :verified_duplicate_channels_must_be_contested, if: -> { verified? }
 
   after_save :register_channel_for_promo, if: :should_register_channel_for_promo
+  after_save :set_channel_banner, if: -> { SiteBanner.find_by(channel_id: self.id).nil? }
```

Comment:
Probably better to let the relation do it for you 
```suggestion
  after_save :set_channel_banner, if: -> { self.site_banner.nil? }
```
not sure if you have to have the `self` there or not